### PR TITLE
AEA-3531 update readme

### PR DESCRIPTION
**Pull request by anthony-nhs**
## Summary

- Routine Change

### Details

- Update readme
- Ensure check-licences is run for all code in the repo 
- Add pull request template
- Add flake8 config
- Update pre-commit hooks
- Update gitpython
- Update authors
- Update licences in package.json to be consistent
- Added workflow to link PR to jira

## Reviews Required

- [x] Dev
- [ ] Test
- [ ] Tech Author
- [ ] Product Owner

## Review Checklist

:information_source: This section is to be filled in by the **reviewer**.

- [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
- [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
- [ ] I have ensured the jira ticket has been updated with the github pull request link

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,10 +207,30 @@ These are used to do common commands
...
+Workflows are in the .github/workflows folder
+
+- `combine-dependabot-prs.yml`: Workflow for combining dependabot pull requests. Runs on demand
+- `delete_old_cloudformation_stacks.yml`: Workflow for deleting old cloud formation stacks. Runs daily
+- `pull_request.yml`: Called when pull request is opened or updated. Calls sam_package_code and sam_release_code to build and deploy the code. Deploys to dev AWS account. The main and sandbox stacks deployed have PR-<PULL REQUEST_ID> in the name
+- `quality_checks.yml`: Runs check-licenses, lint, test and sonarcloud scan agains the repo. Called from pull_request.yml and release.yml
+- `release.yml`: Run when code is merged to main branch or a tag starting v is pushed. Calls sam_package_code and sam_release_code to build and deploy the code.
+- `sam_package_code.yml`: Packages code and uploads to a github artifact for later deployment
+- `sam_release_code.yml`: Release code built by sam_package_code.yml to an environment
+- `pr-link.yaml`: This workflow template links Pull Request's to Jira tickets and runs when a pull request is opened.
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply README.md]: fixed

File: README.md [hunk 2]
```diff
@@ -203,10 +207,30 @@ These are used to do common commands
...
+This .github folder contains workflows and templates related to github
+
+- `pull_request_template.yml`: Template for pull requests.
+
+Workflows are in the .github/workflows folder
+
+- `combine-dependabot-prs.yml`: Workflow for combining dependabot pull requests. Runs on demand
+- `delete_old_cloudformation_stacks.yml`: Workflow for deleting old cloud formation stacks. Runs daily
+- `pull_request.yml`: Called when pull request is opened or updated. Calls sam_package_code and sam_release_code to build and deploy the code. Deploys to dev AWS account. The main and sandbox stacks deployed have PR-<PULL REQUEST_ID> in the name
+- `quality_checks.yml`: Runs check-licenses, lint, test and sonarcloud scan agains the repo. Called from pull_request.yml and release.yml
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply README.md]: fixed

File: README.md [hunk 3]
```diff
@@ -203,10 +207,30 @@ These are used to do common commands
...
+
+This .github folder contains workflows and templates related to github
+
+- `pull_request_template.yml`: Template for pull requests.
+
+Workflows are in the .github/workflows folder
+
+- `combine-dependabot-prs.yml`: Workflow for combining dependabot pull requests. Runs on demand
+- `delete_old_cloudformation_stacks.yml`: Workflow for deleting old cloud formation stacks. Runs daily
+- `pull_request.yml`: Called when pull request is opened or updated. Calls sam_package_code and sam_release_code to build and deploy the code. Deploys to dev AWS account. The main and sandbox stacks deployed have PR-<PULL REQUEST_ID> in the name
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply README.md]: fixed

File: README.md [hunk 4]
```diff
@@ -147,7 +151,7 @@ Any code changes you make are automatically uploaded to AWS while `make sam-sync
 
 ### Pre-commit hooks
 
-Some pre-commit hooks are installed as part of the install above to ensure you can't commit invalid spec changes by accident and to run basic lint checks.
+Some pre-commit hooks are installed as part of the install above to ensure you can't commit invalid changes by accident and by running basic lint checks.
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply README.md]: fixed

**Review by originalphil — APPROVED** [review]

**Pull request merged**